### ISSUE TITLE
polling for jobs removed from the queue does not properly mark them as completed

### ISSUE THREAD

**Issue by hategan** [ExaWorks/psij-python#401]
If a job is attached (or, depending on the polling interval, in normal usage scenarios), a tracked job may be gone from the queue when its status is polled. This is supposed to mark the job as COMPLETED, but there are reports that this is broken.

**Comment by arosen93** [comment]
I can now confirm that this is indeed an issue.

On Perlmutter at NERSC, I ran the following code-block to submit a simple toy job.

```python
from psij import Job, JobAttributes, JobExecutor, JobSpec, ResourceSpecV1
import time

job_executor = JobExecutor.get_instance("slurm")
job = Job(
    JobSpec(
        name="test",
        executable="/bin/date",
        resources=ResourceSpecV1(node_count = 1),
        attributes=JobAttributes(project_name = 'matgen', custom_attributes = {'slurm.constraint': 'cpu', 'slurm.qos': 'debug'}),
        launcher="single",
    )
)
job_executor.submit(job)
native_id = job.native_id
print(native_id)
```

I then manually monitored the queue (i.e. with `squeue`) until the job finished and was no longer visible in the `squeue` history.

Then I ran the following block:
```python
from psij import JobExecutor, Job

job_executor = JobExecutor.get_instance("slurm")
job = Job()
job_executor.attach(job, "13203806") # native ID from above
print(job.status.state)
```

It returned `NEW` every time I printed the state. It also reported an error in `logging` like what follows, but this did not hinder the return of `job.status.state` as `NEW`.

```
Job 651f97ae-c467-4c4d-9edd-f16bf2371d89: failed clean submit script: [Errno 2] No such file or directory: '/global/homes/r/rosen/.psij/work/slurm/651f97ae-c467-4c4d-9edd-f16bf2371d89.job'
```

**Comment by hategan** [comment]
I am getting the error above, and that should be fixed, but if I add a `wait()` or `sleep(3)` after `attach`, the job shows as `COMPLETED`. So I think the main issue goes back to the fact that it takes a bit for the job status to be updated after `attach()`.

**Comment by hategan** [comment]
PR #403 fixes the "failed to clean submit script" error.

**Comment by hategan** [comment]
@Andrew-S-Rosen, can I close this?

**Comment by Andrew-S-Rosen** [comment]
@hategan: Yes, thank you for asking. Go ahead and close the issue.

**Comment by hategan** [comment]
Thank you!

For reference, the "failed to clean submit script" error was fixed by #403 and the COMPLETED state should be reachable with the requisite `wait()` after `attach()`.
